scipy/signal/_signaltools.py

```diff
@@ -1852,9 +1852,33 @@ def medfilt2d(input, kernel_size=3):
     -----
     This is faster than `medfilt` when the input dtype is ``uint8``,
     ``float32``, or ``float64``; for other types, this falls back to
-    `medfilt`; you should use `scipy.ndimage.median_filter` instead as it is
-    much faster.  In some situations, `scipy.ndimage.median_filter` may be
+    `medfilt`. In some situations, `scipy.ndimage.median_filter` may be
     faster than this function.
+    
+    Examples
+    ------
+    Create a zero padded median filtering across an array. 
+    
+    >>> from scipy import signal
+    >>> x = np.random.random((5,5))
```

Comment:
If you don't fix the random seed, the documentation checker will fail to compare the results correctly and will error out. You might want to set a fixed x value to avoid relying on random numbers and will take the example to be more illustrative to show a corner example.

Reply:
As an aside, the refguide checker does something weird: it makes np.random.seed(None) indeed, but patches np.random.default_rng() to use a fixed seed. Not too relevant for this specific example, just something I just stumbled upon.

Reply:
I think `random.random` is discouraged and the seed only works with `default_rng()`. I lost track of the new good practices so just a speculation.

Reply:
The way you write it in the docstring:

```
>>> rng = np.random.default_rng()
>>> x = rng.random((5, 5))
```

The way you run it to generate the example values ([fixed seed from here](https://github.com/scipy/scipy/blob/main/scipy/_lib/_util.py#L549)):

```
>>> rng = np.random.default_rng(1638083107694713882823079058616272161)
>>> x = rng.random((5, 5))
...
```

Reply:
Re: random samples in docstrings, see https://github.com/scipy/scipy/pull/13955#discussion_r623197448 (@tupui's suggestion didn't make it into that PR) and https://github.com/scipy/scipy/issues/15852

Reply:
Yes, as Warren is saying, the overwriting is intentional so we can use `np.random.default_rng()` (which is recommended by NumPy). And I. The background we just replace None with a seed so examples stay the same.

Reply:
Re: inconsistency: Note the comment below this line: https://github.com/scipy/scipy/blob/3280b76e3d16b1c438fb1272dadd64baa5c72984/tools/refguide_check.py#L710

Reply:
I did not do this part but it makes sense to me so to deliberately break seeding with a global state. This way we force best practices in our examples.